### PR TITLE
Milestone 2: Query and Update animals DB

### PR DESCRIPTION
**Pull request by Xghayor**
In this milestone i completed:

- Database transactions.
- Modify and delete data in SQL.
- Complex queries that answer analytical questions.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -6,7 +6,8 @@ CREATE TABLE animals (
     date_of_birth DATE,
     escape_attempts INT,
     neutered BOOLEAN,
-    weight_kg FLOAT
+    weight_kg FLOAT,
+    species VARCHAR(100)
```
Comment (descholar-ceo) [inline schema.sql]: - [ ] It's nice if you do table modifications on a separate statement, don't edit already ran queries:
`ALTER TABLE animals ADD species varchar(255);`

File: queries.sql [hunk 2]
```diff
@@ -19,4 +19,64 @@ SELECT * from animals WHERE name != 'Gabumon';
...
 SELECT *
 FROM animals
-WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+
+UPDATE animals SET species = 'unspecified';
+
```
Comment (descholar-ceo) [inline queries.sql]: - [ ] You are supposed to put this inside a transaction

File: queries.sql [hunk 3]
```diff
@@ -19,4 +19,64 @@ SELECT * from animals WHERE name != 'Gabumon';
...
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species = 'unspecified';
+COMMIT TRANSACTION;
```
Comment (descholar-ceo) [inline queries.sql]: - [ ] You are supposed to run `SELECT species from animals;` in order to check if the changes went through

File: queries.sql [hunk 4]
```diff
@@ -19,4 +19,64 @@ SELECT * from animals WHERE name != 'Gabumon';
...
+
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK TRANSACTION;
```
Comment (descholar-ceo) [inline queries.sql]: - [ ] You are supposed to run a query that reads all animals before and after committing transaction in order to confirm whether the changes worked

File: queries.sql [hunk 5]
```diff
@@ -19,4 +19,64 @@ SELECT * from animals WHERE name != 'Gabumon';
...
+
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT sp1;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO sp1;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
```
Comment (descholar-ceo) [inline queries.sql]: - [ ] You are supposed to commit the transaction after updating where `weight_kg < 0`

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @Xghayor 🤝 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
👍🏽 Professional PR with a title and a description, with screenshots in the description ✔️ 

## Required Changes ♻️
_Check the comments under the review._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema.sql (hunk 1 above)
Comment (Xghayor) [reply schema.sql]: Hi @descholar-ceo ,
Thanks for the feedback, i'll do that.

File: queries.sql (hunk 2 above)
Comment (Xghayor) [reply queries.sql]: Hi @descholar-ceo ,
Thanks for the feedback, i'll do that.

File: queries.sql (hunk 3 above)
Comment (Xghayor) [reply queries.sql]: Hi @descholar-ceo ,
Thanks for the feedback, i'll do that.

File: queries.sql (hunk 4 above)
Comment (Xghayor) [reply queries.sql]: Hi @descholar-ceo ,
Thanks for the feedback, i'll do that.

File: queries.sql (hunk 5 above)
Comment (Xghayor) [reply queries.sql]: Hi @descholar-ceo ,
Thanks for the feedback, i'll do that.

File: queries.sql [hunk 6]
```diff
@@ -19,4 +19,70 @@ SELECT * from animals WHERE name != 'Gabumon';
...
 SELECT *
 FROM animals
-WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+
+ALTER TABLE animals ADD species varchar(255);
+
+
```
Comment (KDlamini) [inline queries.sql]: Well done altering the table properly, as requested by the previous reviewer. 👏🏾 💯 

This line of code should be in the `schema.sql` file. Note that everything that creates or change the tables structure should be saved in the schema.sql file.

1. Kindly move the SQL command on line 25 to its relevant schema file and your project will be ready for approval.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @Xghayor 🙋‍♀️,  

You have done an incredible job implementing some of the changes requested by the previous reviewer. 👏🏾👏🏾👏🏾  
However, there are still some changes that aren't addressed yet. You are almost there! 💪🏾 

## Highlights 🌞 

✔️  Professional pull request
✔️  Correct use of Gitflow 
✔️  All required SQL commands are present

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 6 above)
Comment (Xghayor) [reply queries.sql]: Hi @KDlamini ,
sure, i'll do that, Thanks

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @Xghayor,


68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Great effort on implementing the requested changes🔥 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**